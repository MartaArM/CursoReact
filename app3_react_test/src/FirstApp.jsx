import PropTypes from 'prop-types';

export default function FirstApp( { name } ) { // Es mejor desestructurarlo
    return(
        <>
            <h1>Bienvenido</h1>
            <h2>{ name }</h2>
        </>)
} 

// Decir de qu√© tipo de datos son las propiedades
FirstApp.propTypes = {
    name: PropTypes.string.isRequired // Para hacer la propiedad obligatoria
}

FirstApp.defaultProps = { // Propiedades por defecto
    name: 'No hay nombre'
}