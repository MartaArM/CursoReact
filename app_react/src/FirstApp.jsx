
/* Primera aplicación* /
export default function App() {
    return <h1>Hola Mundo</h1>;
} */

/* ------------------------------------ Agrupar varios elementos ----------------------------* /
export default function App() {
    return(
    <> // Se utiliza esto para agrupar varios elementos
        <h1>Bienvenido</h1>
        <h2>Este es mi nombre</h2>
    </>)
} */

/* --------------------------------Devolver variables al html--------------------------------------* /

function pasoVariable() {
    return "Marta";
}

const miNombre = "Marta";

export default function App() {
    return ( 
        <>
            <h1>Bienvenid@</h1>
            <p>{ miNombre }</p> {/* Le pasamos una variable * /}
            <p>{ pasoVariable() }</p> {/*Se pueden pasar funciones* /}
        </>
    )
}
*/

/* --------------------------- Propiedades que se pueden meter desde el navegador ---------------------------*/

//export default function App(props) {
// export default function App( { name, number } ) { // Es mejor desestructurarlo
//     return(
//         <>
//             <h1>Bienvenido</h1>
//             <h2>{ name }</h2>
//             <p>{ number }</p>
//         </>)
// } 

/* ---------------------------------------Prop-types ---------------------------------------- * /
import PropTypes from 'prop-types';

export default function App( { name, number } ) { // Es mejor desestructurarlo
    return(
        <>
            <h1>Bienvenido</h1>
            <h2>{ name }</h2>
            <p>{ number }</p>
        </>)
} 

// Decir de qué tipo de datos son las propiedades
App.propTypes = {
    name: PropTypes.string.isRequired // Para hacer la propiedad obligatoria
}

App.defaultProps = { // Propiedades por defecto
    name: 'No hay nombre'
}*/

/* --------------------------------------- Agregar eventos a botones----------------------------------------* /
let contador = 0;

// const imprimir = (event)  => {
//     contador+=1;
//     console.log("Imprimir " + contador);
//     console.log(event); // Podemos mandar la información del evento
// }

const imprimir = (event, value)  => {
    console.log(event); // Podemos mandar la información del evento
    console.log(value);
}

export default function App() {

    return ( 
        <>
            <h1>Bienvenid@</h1>
            {/* <button onClick = { imprimir }> Boton </button> * /}
             <button onClick = { (event) => imprimir(event, "hola") }> Boton </button> {/*Esto se usa si se quiere pasar algun valor a la función * /}
        </>
    )
}
*/
/* --------------------------------------- Hooks ----------------------------------------*/

import {useState} from 'react';

export default function App() {

    const [contador, setCounter] = useState(0); // Esto es el hook. Ponemos contador a 0

    const aumentarContador = () => {
        setCounter(contador+1); // No lo podemos hacer directamente. Se hace con la funcion setCounter
    }
    return ( 
        <>
            <h1>Bienvenid@</h1>
            <button onClick = { aumentarContador }> Boton </button>
            <h2>{ contador }</h2>
        </>
    )
}