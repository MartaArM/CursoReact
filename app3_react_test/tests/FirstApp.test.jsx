import { render } from '@testing-library/react'
import FirstApp from '../src/FirstApp'

describe('Prueba con FirstApp', () => { 
    
    test('Primera prueba' , () => {

            const { container } = render (<FirstApp name='Hola'/>);

            expect( container ).toMatchSnapshot();
 
    })


})