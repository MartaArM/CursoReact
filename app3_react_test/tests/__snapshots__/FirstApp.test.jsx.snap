// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con FirstApp Primera prueba 1`] = `
<div>
  <h1>
    Bienvenido
  </h1>
  <h2>
    Hola
  </h2>
</div>
`;
